xonsh/built_ins.py

```diff
@@ -637,15 +637,41 @@ def run_subproc(cmds, captured=True):
             hist.last_cmd_out = output
     if not prev_is_proxy and hist.last_cmd_rtn > 0 and ENV.get('RAISE_SUBPROC_ERROR'):
         raise CalledProcessError(hist.last_cmd_rtn, aliased_cmd, output=output)
-    if captured:
+    if captured=='stdout':
         return output
+    elif captured=='object':
+        return CompletedProcess(prev_proc, output)
 
 
-def subproc_captured(*cmds):
+class CompletedProcess:
```

Comment:
Yeah, I pretty much agree that the interface should be the same, or at least that the interface here should be a superset of the interface of subprocess.